packages/analytics-element/src/tracked-element.js

```diff
@@ -0,0 +1,23 @@
+export default {
+  trackClick(e) {
+    const event = e.currentTarget.dataset.eventClickTracking;
+    this.dispatchEvent(new CustomEvent('trackClick', {
+      bubbles: true,
+      composed: true,
+      detail: {
+        event
+      },
+    }));
+  },
+
+  trackSubmit(e) {
+    const event = e.currentTarget.dataset.eventSubmitTracking;
+    this.dispatchEvent(new CustomEvent('trackSubmit', {
+      bubbles: true,
+      composed: true,
+      detail: {
+        event
+      },
+    }));
+  }
+};
```

Comment:
a few questions, comments come to mind with this file, they may compound on each other.  Let me list as I think, perhaps we can solidify some of my thoughts together:
- seems like each method is the same inside..
- can we whitelist these events, do we even need a white list?
- I can see the value in an explicit list, that way we can catch all the various events - what will this list look like?
- for "generic" purposes, should we extract these event names as constants that are defaults but can be overridden?


archive.org events look like this: `Foo|Bar` - I don't think the `example-component` should know this syntax - we should abstract this up one level into a helper function of some kind inside the tracking functionality.  I think this will help generalize the behavior on the consumption level.

ex.
`this.trackEvent({ event: 'Submit', element: 'ExampleTracker', details: 'foo' })`
where track event uses archive.org specific transformer (as default or can take another transformer) to make string and then fire.